src/Simplex/FileTransfer/Description.hs

```diff
@@ -204,17 +225,33 @@ validateFileDescription fd@FileDescription {size, chunks}
     chunksSize = fromIntegral . foldl' (\s FileChunk {chunkSize} -> s + unFileSize chunkSize) 0
 
 encodeFileDescription :: FileDescription p -> YAMLFileDescription
-encodeFileDescription FileDescription {party, size, digest, key, nonce, chunkSize, chunks} =
+encodeFileDescription FileDescription {party, size, digest, key, nonce, chunkSize, chunks, redirect} =
   YAMLFileDescription
     { party = toFileParty party,
       size = B.unpack $ strEncode size,
       digest,
       key,
       nonce,
       chunkSize = B.unpack $ strEncode chunkSize,
-      replicas = encodeFileReplicas chunkSize chunks
+      replicas = encodeFileReplicas chunkSize chunks,
+      redirect
     }
 
+data FileDescriptionURI = FileDescriptionURI ServiceScheme (ValidFileDescription 'FRecipient)
+  deriving (Eq, Show)
+
+instance StrEncoding FileDescriptionURI where
+  strEncode (FileDescriptionURI ss vfd) = mconcat [strEncode ss, "/file#?d=", urlEncode True $ strEncode vfd]
+  strP = do
+    ss <- strP
+    _ <- "/file#?d="
+    dUrl <- A.takeByteString
```

Comment:
also, I think we need to do it via query string parser, as we did with other URIs, otherwise we won't be able to extend it without breaking it for the old clients.